scipy/stats/_wilcoxon.py

```diff
@@ -97,38 +97,35 @@ def _wilcoxon_iv(x, y, zero_method, correction, alternative, method, axis):
         raise ValueError(message)
 
     if not isinstance(method, stats.PermutationMethod):
-        methods = {"auto", "approx", "exact"}
+        methods = {"auto", "asymptotic", "exact"}
         message = (f"`method` must be one of {methods} or "
                    "an instance of `stats.PermutationMethod`.")
         if method not in methods:
             raise ValueError(message)
-    output_z = True if method == 'approx' else False
-
-    # logic unchanged here for backward compatibility
-    n_zero = np.sum(d == 0, axis=-1)
-    has_zeros = np.any(n_zero > 0)
-    if method == "auto":
-        if d.shape[-1] <= 50 and not has_zeros:
-            method = "exact"
-        else:
-            method = "approx"
+    output_z = True if method == 'asymptotic' else False
 
+    # For small samples, we decide later whether to perform an exact test or a
+    # permutation test. The reason is that the presence of ties is not
+    # known at the input validation stage.
     n_zero = np.sum(d == 0)
+    if method == "auto" and d.shape[-1] > 50:
+        method = "asymptotic"
+
     if n_zero > 0 and method == "exact":
-        method = "approx"
         warnings.warn("Exact p-value calculation does not work if there are "
-                      "zeros. Switching to normal approximation.",
+                      "zeros. Consider using method `asymptotic` or "
+                      "`stats.PermutationMethod`",
                       stacklevel=2)
 
-    if (method == "approx" and zero_method in ["wilcox", "pratt"]
+    if (method == "asymptotic" and zero_method in ["wilcox", "pratt"]
```

Comment:
Does this condition make sense? If we use asymptotic with `d = np.zeros(10)`, we raise an error. If `d = np.zeros((10, 10))`,  one gets an array with statistic = 0, pvalue = nan.

Reply:
I don't remember, but I might have done it this way when I added native `axis` support to preserve existing behavior for 1-D arrays but not let a single slice of zeros in an N-D array cause the whole calculation to error out. I would just assume that we always return NaN - an appropriate result when there is not a numerical result - and document this condition*if* it is so confusing to the user why they can't get a p-values for degenerate data.

Reply:
I agree, so you suggest to remove this check? I think it would be consistent with the behavior of N-D arrays. There will be a warning indicating that the input is not good and we can document it.

Reply:
That sounds good if the warning you're referring to is the standard divide by zero warning (not custom), so no accommodations are made in the code.

Reply:
yes, exactly. the warning emitted w/o any changes to our code is "RuntimeWarning: invalid value encountered in divide z = (r_plus - mn) / se"